### DISCUSSION TITLE
Dealing with multiple redis services

### DISCUSSION THREAD

**Discussion by kaibagley** [immich-app/immich#10913] [Q&A] [answered]
Hey guys, 

Thought I'd as here as I'm sure someone's dealt with this, but following the docs I end up with a compose.yml with a redis service. However I already have a redis service for Authentik. Renaming the service for Immich by changing `redis:` to `immich-redis:`, and likewise in the `depends_on:` in the `immich_server` service. Results in errors in `immich_server`: 

Docker logs gives:
```
Error: getaddrinfo ENOTFOUND redis
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26) {
  errno: -3007,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'redis'
}
```

So i must be missing somewhere else I need to change the service name. Or, maybe there's some other solution here that others use to deal with multiple services with the same name.

Thanks!

**Comment by bo0tzz** [comment, marked as answer]
Services in different compose files are entirely isolated, you shouldn't need to change anything.

**Reply by kaibagley** [reply]
I was thinking about that but thought it'd be better to keep everything in one file, do you think it's best practice to keep the Immich compose file separate from the rest of my home server?

**Reply by bo0tzz** [reply]
Yes, you should have a separate compose file for each service. If they need to communicate in some way (eg for your reverse proxy to reach them) you can do that with a separate docker network.

**Reply by kaibagley** [reply]
Alright perfect, thanks so much for your help, and for the super fast replies!!
